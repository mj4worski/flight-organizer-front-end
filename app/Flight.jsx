import React from 'react'

const Flight = (flight) => {
    return (
        <section>
            <h1>Znaleziony lot to:</h1>
            <p>Skad : {flight.flight.from}</p>
            <p>Dokad : {flight.flight.to}</p>
        </section>
    )
};

export default Flight;