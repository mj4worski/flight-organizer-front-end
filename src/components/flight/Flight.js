// @flow
import React from 'react';
import type { FlightType } from '../../api/types';

type FlightComponentType = {
    flight: FlightType,
    onReservationClick: Function,
}

export default ({ flight, onReservationClick }: FlightComponentType) => (
  <section>
    <h1>Znaleziony lot to:</h1>
    <p>Skad : {flight.departureFrom}</p>
    <p>Dokad : {flight.arrivalTo}</p>
    <button onClick={() => onReservationClick(flight)}>Rezerwacja</button>
  </section>
);
