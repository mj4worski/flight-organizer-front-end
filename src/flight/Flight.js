import React from 'react'

const Flight = ({flight, onReservationClick}) => {
    return (
        <section>
            <h1>Znaleziony lot to:</h1>
            <p>Skad : {flight.departureFrom}</p>
            <p>Dokad : {flight.arrivalTo}</p>
            <button onClick={(e) => onReservationClick(flight)}>Rezerwacja</button>
        </section>
    )
};

export default Flight;