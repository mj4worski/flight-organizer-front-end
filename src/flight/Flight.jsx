import React from 'react'

const Flight = ({from ,to ,date}) => {
    return (
        <section>
            <h1>Znaleziony lot to:</h1>
            <p>Skad : {from}</p>
            <p>Dokad : {to}</p>
            {/*<p>Kiedy : {date.dayOfWeek} {date.month} {date.year}</p>*/}
        </section>
    )
};

export default Flight;