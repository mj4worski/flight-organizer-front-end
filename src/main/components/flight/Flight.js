import React from 'react';
import PropTypes from 'prop-types';

const Flight = ({ flight }) => (
  <section>
    <h1>Znaleziony lot to:</h1>
    <p>Skad : {flight.departureFrom}</p>
    <p>Dokad : {flight.arrivalTo}</p>
    <button onClick={() => {}}>Rezerwacja</button>
  </section>
);

Flight.propTypes = {
  flight: PropTypes.object.isRequired,
};

export default Flight;
