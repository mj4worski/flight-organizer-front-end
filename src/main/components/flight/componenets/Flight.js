import React from 'react';
import PropTypes from 'prop-types';

const Flight = ({ flight }) => (
  <section>
    <h1>Founded flights:</h1>
    <p>From : {flight.departureFrom}</p>
    <p>To : {flight.arrivalTo}</p>
    <button onClick={() => {}}>Book It</button>
  </section>
);

Flight.propTypes = {
  flight: PropTypes.shape({
    departureFrom: PropTypes.string,
    arrivalTo: PropTypes.string,
  }).isRequired,
};

export default Flight;
