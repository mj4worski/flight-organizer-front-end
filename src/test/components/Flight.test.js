import React from 'react';
import renderer from 'react-test-renderer';
import Flight from '../../main/components/flight/Flight';

it('should render Flight component with arrivalTo and departureFrom props', () => {
  const flight = {
    arrivalTo: 'Paris',
    departureFrom: 'Barcelona',
  };
  const component = renderer.create(
    <Flight flight={flight} />,
    );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
