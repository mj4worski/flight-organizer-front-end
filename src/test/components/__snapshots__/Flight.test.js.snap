// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Flight component with arrivalTo and departureFrom props 1`] = `
<section>
  <h1>
    Founded flights:
  </h1>
  <p>
    From : 
    Barcelona
  </p>
  <p>
    To : 
    Paris
  </p>
  <button
    onClick={[Function]}
  >
    Book It
  </button>
</section>
`;
